q2_phylogenomics/_assemble.py

```diff
@@ -0,0 +1,121 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import tempfile
+import skbio
+import pandas as pd
+import os.path
+import hashlib
+
+from q2_types.per_sample_sequences import (
+    SingleLanePerSamplePairedEndFastqDirFmt)
+from q2_types.bowtie2 import Bowtie2IndexDirFmt
+from q2_types.feature_data import (DNASequencesDirectoryFormat,
+                                   DNAFASTAFormat)
+
+from ._util import run_command
+from ._format import (
+        BAMFilesDirFmt, PileUpFilesDirFmt,
+        BAMFormat, PileUpTSVFormat)
+
+
+def map_paired_reads(demux: SingleLanePerSamplePairedEndFastqDirFmt,
+                     reference: Bowtie2IndexDirFmt,
+                     mismatches_per_seed: int = 1,
```

Comment:
want to make a dict of default params for each method? Then I can easily import that dict for setting defaults in the pipeline.